javatests/arcs/core/host/HandleAdapterTest.kt

```diff
@@ -109,13 +110,92 @@ class HandleAdapterTest {
         assertThat(writeOnlyHandle).isNotInstanceOf(ReadWriteCollectionHandle::class.java)
     }
 
+    @Test
+    fun singletonHandleAdapter_onUpdateTest() = runBlockingTest {
+        val handle = manager.createSingletonHandle(
+            HandleMode.ReadWrite,
+            READ_WRITE_HANDLE,
+            Person,
+            STORAGE_KEY_ONE,
+            Person.SCHEMA
+        ) as ReadWriteSingletonHandle<Person>
+
+        var x = 0
+        handle.onUpdate { p ->
+            if(p?.name == "Eliza Hamilton") {
+                x = 1
```

Comment:
rather than `x = 1`, it's probably better to do `x++`.. that way if the callback gets triggered more than once, the test will fail